hs_core/views/serializers.py

```diff
@@ -0,0 +1,75 @@
+from rest_framework import serializers
+from .utils import authorize, validate_json, validate_user_name, validate_group_name
+from hs_core.hydroshare import utils
+
+
+class ResourceUpdateRequestValidator(serializers.Serializer):
+    title = serializers.CharField(required=False)
+    metadata = serializers.CharField(validators=[validate_json], required=False)
+    # TODO: use a custom validator for user names and group names
+    user_choices = [user.username for user in User.objects.all()]
+    group_choices = [grp.name for grp in Group.objects.all()]
+    edit_users = serializers.MultipleChoiceField(choices=user_choices, required=False)
+    edit_groups = serializers.MultipleChoiceField(choices=group_choices, required=False)
+    view_users = serializers.MultipleChoiceField(choices=user_choices, required=False)
+    view_groups = serializers.MultipleChoiceField(choices=group_choices, required=False)
+
+
+class ResourceCreateRequestValidator(ResourceUpdateRequestValidator):
+    resource_type = serializers.ChoiceField(
+            choices=zip(
+                [x.__name__ for x in hydroshare.get_resource_types()],
+                [x.__name__ for x in hydroshare.get_resource_types()]
+            ), default='GenericResource')
+
+
+class ResourceListRequestValidator(serializers.Serializer):
+    group = serializers.CharField(min_length=1, required=False, validators=[validate_group_name])
+    user = serializers.CharField(min_length=1, required=False, validators=[validate_user_name])
+    owner = serializers.CharField(min_length=1, required=False, validators=[validate_user_name])
+    from_date = serializers.DateField(required=False, default=None)
+    to_date = serializers.DateField(required=False, default=None)
+    start = serializers.IntegerField(required=False, default=None)
+    count = serializers.IntegerField(required=False, default=None)
+    metadata = serializers.CharField(min_length=1, required=False, validators=[validate_json])
+    full_text_search = serializers.CharField(required=False)
+    edit_permission = serializers.BooleanField(required=False, default=False)
+    published = serializers.BooleanField(required=False, default=False)
+    resource_types = utils.get_resource_types()
+    resource_type_choices = [rtype.__name__ for rtype in resource_types]
+    types = serializers.MultipleChoiceField(choices=resource_type_choices, required=False, default=None)
+
+
+class ResourceListItemSerializer(serializers.Serializer):
+    resource_type = serializers.CharField(max_length=100)
+    resource_title = serializers.CharField(max_length=200)
+    resource_id = serializers.CharField(max_length=100)
+    creator = serializers.CharField(max_length=100)
+    date_created = serializers.DateTimeField(format='%d-%m-%Y')
+    date_last_updated = serializers.DateTimeField(format='%d-%m-%Y')
+    sharing_status = serializers.CharField(max_length=20)
+    bag = serializers.URLField()
+    science_metadata = serializers.URLField()
```

Comment:
I would recommend that any URL field be named ending with _url to distinguish from fields that contain data directly.